universum/modules/automation_server/teamcity_server.py

```diff
@@ -87,18 +88,20 @@ def __init__(self, *args, **kwargs):
             of the top-level project on the TeamCity.
             """)
 
-        self.tc_build_link = self.settings.server_url + "/viewLog.html?tab=buildLog&buildId=" + self.settings.build_id
-        self.tc_artifact_link = self.settings.server_url + "/repository/download/" + \
-            self.settings.configuration_id + "/" + self.settings.build_id + ":id/"
+        server_url: str = self.settings.server_url
+        build_id: str = self.settings.build_id
+        conf_id: str = self.settings.configuration_id
+        self.tc_build_link: str = server_url + "/viewLog.html?tab=buildLog&buildId=" + build_id
+        self.tc_artifact_link: str = server_url + "/repository/download/" + conf_id + "/" + build_id + ":id/"
 
-    def report_build_location(self):
+    def report_build_location(self) -> str:
         return "Here is the link to TeamCity build: " + self.tc_build_link
 
-    def artifact_path(self, local_artifacts_dir, item):
+    def artifact_path(self, local_artifacts_dir: str, item: str) -> str:
         return self.tc_artifact_link + item
 
-    def add_build_tag(self, tag):
+    def add_build_tag(self, tag: str) -> str:
         return requests.post("%s/httpAuth/app/rest/builds/id:%s/tags" %
                              (self.settings.server_url, self.settings.build_id),
                              auth=(self.settings.user_id, self.settings.passwd),
-                             data=tag, headers={'Content-Type': 'text/plain'}, verify=False)
+                             data=tag, headers={'Content-Type': 'text/plain'}, verify=False).text
```

Comment:
do note ".text" here. It seems to me that str is expected, not Response

Reply:
Where did the '.text' come from? It returned full Response before this change

Reply:
I wasn't sure about this one - which is exactly why I added this comment.